ank-core/src/main/kotlin/io/kategory/ank/interpreter.kt

```diff
@@ -118,37 +126,58 @@ fun extractCodeImpl(source: String, tree: ASTNode): ListKW<Snippet> {
     return sb.k()
 }
 
-fun compileCodeImpl(origin: File, snippets: ListKW<Snippet>, classpath: ListKW<String>): CompiledMarkdown {
-    println(":runAnk -> Compiling: $origin")
-    val classLoader = URLClassLoader(classpath.map { URL(it) }.ev().list.toTypedArray())
-    val seManager = ScriptEngineManager(classLoader)
-    val cachedEngines: ConcurrentHashMap<String, ScriptEngine> = ConcurrentHashMap()
-    val evaledSnippets = snippets.list.map { snippet ->
-        val result = Try {
-            val engine = cachedEngines.get(snippet.lang)
-            val resolvedEngine = if (engine != null) engine else {
-                seManager.getEngineByExtension(extensionMappings.getOrDefault(snippet.lang, "kts"))!!
+
+fun compileCodeImpl(snippets: Map<File, ListKW<Snippet>>, classpath: ListKW<String>): ListKW<CompiledMarkdown> {
+    println(":runAnk -> started compilation")
+    val IOPool = object : CoroutineDispatcher() {
```

Comment:
I'm quite sure that kotlinx.coroutines provides a better abstraction for a weighted ThreadPool.